mlir/test/Dialect/Rock/test_transformation_maps_utils.mlir

```diff
@@ -0,0 +1,119 @@
+
+// CHECK-LABEL: test1 transform_ex4
+// CHECK-NEXT: #rock.transform_map<affine_map<(d0) -> (d0, 0)> by [<PassThrough ["A"] at [0] -> ["A"] at [0]>, <ConstDim{0, 1} [] at [] -> ["B"] at [1]>] bounds = [1] -> [1, 1024]>
+// CHECK-NEXT: #rock.transform_map<affine_map<(d0, d1) -> (d0, d1)> by [<PassThrough ["A"] at [0] -> ["A"] at [0]>, <PassThrough ["B"] at [1] -> ["B"] at [1]>] bounds = [1, 1024] -> [1, 1024]>
+
+// CHECK-LABEL: test2 transform_ex4
+// CHECK-NEXT: #rock.transform_map<affine_map<() -> (0)> by [<ConstDim{0, 1} [] at [] -> ["B"] at [0]>] bounds = [] -> [1024]>
+// CHECK-NEXT: #rock.transform_map<affine_map<(d0) -> (d0)> by [<PassThrough ["B"] at [0] -> ["B"] at [0]>] bounds = [1024] -> [1024]>
+
+func.func @transform_ex4(%arg0: memref<1x64x1024xf32>) -> memref<1x64xf32>
+  attributes {
+    remove_dims_by_names = {
+      test0 = ["A"],
+      test1 = ["C"],
+      test2 = ["A", "C"]
+    }
+  } {
+  %0 = rock.transform %arg0 by <affine_map<(d0, d1, d2) -> (d0, d2, d1)> by [<PassThrough ["A"] at [0] -> ["A"] at [0]>, <PassThrough ["B", "C"] at [1, 2] -> ["B", "C"] at [2, 1]>] bounds = [1, 1024, 64] -> [1, 64, 1024]> : memref<1x64x1024xf32> to memref<1x1024x64xf32>
+  %1 = rock.transform %0 by <affine_map<(d0, d2) -> (d0, 0, d2)> by [<PassThrough ["A"] at [0] -> ["A"] at [0]>, <ConstDim{0, 1} [] at [] -> ["B"] at [1]>, <PassThrough ["C"] at [1] -> ["C"] at [2]>] bounds = [1, 64] -> [1, 1024, 64]> : memref<1x1024x64xf32> to memref<1x64xf32>
+  return %1 : memref<1x64xf32>
+}
+
+// CHECK-LABEL: test0 transform_ex5
+// CHECK-NEXT: #rock.transform_map<affine_map<(d0, d1, d2) -> (d0, d1, d2)> by [<PassThrough ["A", "B", "C"] at [0, 1, 2] -> ["A", "B", "C"] at [0, 1, 2]>] bounds = [1, 1024, 64] -> [1, 1024, 64]>
+// CHECK-NEXT: #rock.transform_map<affine_map<(d0, d1, d2) -> (d0, d2, d1)> by [<PassThrough ["A"] at [0] -> ["A"] at [0]>, <PassThrough ["B", "C"] at [1, 2] -> ["B", "C"] at [2, 1]>] bounds = [1, 1024, 64] -> [1, 64, 1024]>
+
+// CHECK-LABEL: test1 transform_ex5
+// CHECK-NEXT: #rock.transform_map<affine_map<(d0) -> ()> by [<AddDim{8} ["X"] at [0] -> [] at []>] bounds = [8] -> []>
+
+func.func @transform_ex5(%arg0: memref<1x64x1024xf32>) -> memref<1x8x1024x64xf32>
+  attributes {
+    remove_dims_by_names = {
+      test0 = ["X"],
+      test1 = ["A", "B", "C"]
+    }
+  } {
+  %0 = rock.transform %arg0 by <affine_map<(d0, d1, d2) -> (d0, d2, d1)> by [<PassThrough ["A"] at [0] -> ["A"] at [0]>, <PassThrough ["B", "C"] at [1, 2] -> ["B", "C"] at [2, 1]>] bounds = [1, 1024, 64] -> [1, 64, 1024]> : memref<1x64x1024xf32> to memref<1x1024x64xf32>
+  %1 = rock.transform %0 by <affine_map<(d0, d1, d2, d3) -> (d0, d2, d3)> by [<AddDim{8} ["X"] at [1] -> [] at []>, <PassThrough ["A", "B", "C"] at [0, 2, 3] -> ["A", "B", "C"] at [0, 1, 2]>] bounds = [1, 8, 1024, 64] -> [1, 1024, 64]> : memref<1x1024x64xf32> to memref<1x8x1024x64xf32>
```

Comment:
This is a very weird structure - assuming this even needs to be a test pass.

Why not

```
%0 = rock.transform ...
%1 = rock.transform ...
%2 = "drop_dimensions"(%1) {names = ["A"]} : (T) -> (T)
return %2
```
and then "drop_dimensions" is the signal to run the function you're trying to test

Reply:
So, `drop_dimensions` is an Op, right. So, I would need to handle it.

I define tests (i.e., which dimensions to remove) in each function attribute. For example, 

https://github.com/ROCm/rocMLIR/blob/9ed596f77f406852aceab376ccc259d226008e31/mlir/test/Dialect/Rock/test_transformation_maps_utils.mlir#L109-L115

In this particular example, `func.func @transform_ex5` is going to be used in two tests. The expected results of the tests are given about the function - i.e.,

https://github.com/ROCm/rocMLIR/blob/9ed596f77f406852aceab376ccc259d226008e31/mlir/test/Dialect/Rock/test_transformation_maps_utils.mlir#L102-L107

I didn't quite get your point about *"very weird structure"* and *"assuming this even needs to be a test pass"*

Reply:
done